src/runtime/basic-runtime-test.js

```diff
@@ -261,6 +263,116 @@ describes.realWin('BasicRuntime', (env) => {
         'publisherProvidedId'
       );
     });
+
+    [
+      {
+        isAccessibleForFree: true,
+        isPartOfProductId: 'publication:openaccess',
+        isAccessibleFromProductId: true,
+      },
+      {
+        isAccessibleForFree: true,
+        isPartOfProductId: 'publication:notopen',
+        isAccessibleFromProductId: true,
+      },
+      {
+        isAccessibleForFree: false,
+        isPartOfProductId: 'publication:openaccess',
+        isAccessibleFromProductId: false,
+      },
+      {
+        isAccessibleForFree: false,
+        isPartOfProductId: 'publication:notopen',
+        isAccessibleFromProductId: false,
+      },
+      {
+        isAccessibleForFree: undefined,
+        isPartOfProductId: 'publication:openaccess',
+        isAccessibleFromProductId: true,
+      },
+      {
+        isAccessibleForFree: undefined,
+        isPartOfProductId: 'publication:notopen',
+        isAccessibleFromProductId: false,
+      },
+    ].forEach(
+      ({isAccessibleForFree, isPartOfProductId, isAccessibleFromProductId}) => {
+        it(`When isAccessibleForFree=${isAccessibleForFree}, isPartOfProductId=${isPartOfProductId}, expect writePageConfig to be called with isAccessibleForFree=${isAccessibleFromProductId}`, async () => {
```

Comment:
nit: Can you write these in the typical `it` style? eg, it('writes page config with isAccesibleForFree=x when isAccessibleForFree=y and isPartOfProductId=z')

Reply:
+1

Reply:
Done!